TAO/TAO_IDL/fe/idl.ypp

```diff
@@ -3889,6 +3892,138 @@ enumerator :
         }
         ;
 
+map_type_spec :
+        IDL_MAP
+        '<'
+        map_type
+        ','
+        map_type
+        '>'
+        {
+
+          AST_Map *map = 0;
+          Decl_Annotations_Pair *key_type = $3;
+          Decl_Annotations_Pair *val_type = $5;
+
+          /*
+           * Remove sequence marker from scopes stack.
+           */
+          if (idl_global->scopes ().top () == 0)
+            {
+             idl_global->scopes ().pop ();
+            }
+
+          UTL_Scope *s = idl_global->scopes ().top_non_null ();
+
+          /*
+           * Create a node representing a sequence.
+           */
+          if (key_type && val_type)
+            {
+              AST_Type *ktp = dynamic_cast<AST_Type*> (key_type->decl);
+              AST_Type *vtp = dynamic_cast<AST_Type*> (val_type->decl);
+
+              if (ktp == 0 || vtp == 0)
+                {
+                  ; // Error will be caught in FE_Declarator.
+                }
+              else
+                {
+                  Identifier id ("map");
+                  UTL_ScopedName sn (&id, 0);
+                  ACE_CDR::ULong bound = 0UL;
+
+                  map =
+                    idl_global->gen ()->create_map (
+                        idl_global->gen ()->create_expr (
+                                                bound,
+                                                AST_Expression::EV_ulong
+                                              ),
+                        ktp,
+                        vtp,
+                        &sn,
+                        s->is_local (),
+                        s->is_abstract ()
+                      );
+                  // map->base_type_annotations (*type_annotations);
+
+                  idl_global->err ()->anonymous_type_diagnostic ();
+                }
+            }
+
+          $$ = map;
```

Comment:
```suggestion
          delete key_type->annotations;
          delete value_type->annotations;
          $$ = map;
```

Annotations are always copied and the originals have to be deleted or they'll leak.

The same thing for the bounded version.